velox/functions/lib/MysqlDateTime.cpp

```diff
@@ -0,0 +1,160 @@
+        case 'd':
+          builder.appendDayOfMonth(2);
+          break;
+        case 'e':
+          builder.appendDayOfMonth(1);
+          break;
+        case 'f':
+          builder.appendFractionOfSecond(6);
+          break;
+        case 'H':
+          builder.appendHourOfDay(2);
+          break;
+        case 'h':
+        case 'I':
+          builder.appendClockHourOfHalfDay(2);
+          break;
+        case 'i':
+          builder.appendMinuteOfHour(2);
+          break;
+        case 'j':
+          builder.appendDayOfYear(3);
+          break;
+        case 'k':
+          builder.appendHourOfDay(1);
+          break;
+        case 'l':
+          builder.appendClockHourOfHalfDay(1);
+          break;
+        case 'M':
+          builder.appendMonthOfYearText(4);
+          break;
+        case 'm':
+          builder.appendMonthOfYear(2);
+          break;
+        case 'p':
+          builder.appendHalfDayOfDay();
+          break;
+        case 'r':
+          builder.appendClockHourOfHalfDay(2);
+          builder.appendLiteral(":");
+          builder.appendMinuteOfHour(2);
+          builder.appendLiteral(":");
+          builder.appendSecondOfMinute(2);
+          builder.appendLiteral(" ");
+          builder.appendHalfDayOfDay();
+          break;
+        case 'S':
+        case 's':
+          builder.appendSecondOfMinute(2);
+          break;
+        case 'T':
+          builder.appendHourOfDay(2);
+          builder.appendLiteral(":");
+          builder.appendMinuteOfHour(2);
+          builder.appendLiteral(":");
+          builder.appendSecondOfMinute(2);
+          break;
+        case 'v':
+          builder.appendWeekOfWeekYear(2);
+          break;
+        case 'W':
+          builder.appendDayOfWeekText(4);
+          break;
+        case 'x':
+          builder.appendWeekYear(4);
+          break;
+        case 'Y':
+          builder.appendYear(4);
+          break;
+        case 'y':
+          builder.appendYear(2);
+          break;
+        case '%':
+          builder.appendLiteral("%");
+        case 'D':
+        case 'U':
+        case 'u':
+        case 'V':
+        case 'w':
+        case 'X':
+          VELOX_UNSUPPORTED("Specifier {} is not supported.", *tokenEnd)
+          break;
+        default:
+          builder.appendLiteral(tokenEnd, 1);
+          break;
+      }
+      ++tokenEnd;
+    } else if (*tokenEnd == '\'') { // quoted/escaped literal, no need for
+                                    // interpretation
+      ++tokenEnd;
+      while (tokenEnd < end && *tokenEnd != '\'') {
+        ++tokenEnd;
+      }
+      if (tokenEnd == end) {
+        VELOX_USER_FAIL("Unmatched single quote in pattern");
+      }
+      ++tokenEnd; // Skip ending quote
+      if (tokenEnd - cur == 2) { // escaped single quote
+        builder.appendLiteral("'");
+      } else { // quoted literal
+        builder.appendLiteral(cur + 1, tokenEnd - cur - 2);
+      }
+    } else { // unquoted literal
```

Comment:
So, what is the difference between quoted and unquoted literals?
Is it that in quotes I can use the "key_symbols"?

Reply:
Yeah, so we don't interpret contents inside quoted literals-> %Y-'%Y' will be 2022-%Y